scalameta/io/js/src/main/scala/java/io/File.scala

```diff
@@ -0,0 +1,55 @@
+package java.io
+
+import java.net.URI
+import java.nio.file.Path
+import scala.meta.internal.io.FileIO
+import scala.meta.internal.io.JSFs
+import scala.meta.internal.io.JSPath
+import scala.meta.internal.io.JSStats
+import scala.meta.internal.io.NodeNIOPath
+import scala.meta.internal.io.PathIO
+import scala.meta.internal.io.PlatformFileIO
+
+// obtained implementation by experimentation on the JDK.
```

Comment:
That's perfectly fine. It means someone reverse-engineered the spec by observing the behavior, and then reimplemented that spec. That's the very best incarnation of a clean room implementation.